apps/accounts/migrations/0055_populate_existing_permissions.py

```diff
@@ -0,0 +1,127 @@
+        codename=manage_datacenter.codename,
+        name=manage_datacenter.description,
+        content_type=ContentType.objects.get_for_model(Datacenter),
+    )
+    hp_perm, _ = Permission.objects.using(db_alias).get_or_create(
+        codename=manage_provider.codename,
+        name=manage_provider.description,
+        content_type=ContentType.objects.get_for_model(Hostingprovider),
+    )
+
+    return (hp_perm, dc_perm)
+
+
+def set_admin_global_perms(apps, schema_editor):
+    """
+    Assign global permissions (for all Hostingprovider and Datacenter objects)
+    for all members of the "admin" group.
+    """
+    db_alias = schema_editor.connection.alias
+    admin_group = Group.objects.using(db_alias).filter(name="admin").get()
+
+    for perm in _get_managed_perms(apps, db_alias):
+        assign_perm(perm, admin_group)
+
+
+def remove_admin_global_perms(apps, schema_editor):
+    """
+    Remove global permissions (for all Hostingprovider and Datacenter objects)
+    from all members of the "admin" group.
+    """
+    db_alias = schema_editor.connection.alias
+    admin_group = Group.objects.using(db_alias).filter(name="admin").get()
+
+    for perm in _get_managed_perms(apps, db_alias):
+        remove_perm(perm, admin_group)
+
+
+def _manage_permission(apps, schema_editor, manage_func):
+    """
+    Assign/Remove object-level permissions for existing
+    Hostingprovider and Datacenter objects.
+
+    Parameter `manage_func` is either: assign_perm or remove_perm from guardian.shortcuts
+    """
+    db_alias = schema_editor.connection.alias
+
+    hp_perm, dc_perm = _get_managed_perms(apps, db_alias)
+
+    # to query the objects eligible for assigning permissions we need the historical model
+    # so that we can use queries based on fields present at the moment of the migration
+    Datacenter_historical_model = apps.get_model("accounts", "Datacenter")
+    Hostingprovider_historical_model = apps.get_model("accounts", "Hostingprovider")
+
+    # retrieve data based on queryset using historical model to access "created_by" field
+    eligible_pairs_dc = [
+        (dc.id, dc.created_by.id)
+        for dc in Datacenter_historical_model.objects.using(db_alias).filter(
+            created_by__isnull=False
+        )
+    ]
+    eligible_pairs_hp = [
+        (hp.id, hp.created_by.id)
+        for hp in Hostingprovider_historical_model.objects.using(db_alias).filter(
+            created_by__isnull=False
+        )
+    ]
+    # Get current models to retrieve the objects for assigning permissions.
+    # This is dangerous! Future versions of django-guardian might change the schema
+    # and this migration might fail when ran from scratch. If this happens:
+    # - pin down the version of django-guardian to the latest working version
+    #   so that migrations work for tests
+    # - for production / staging it's a non-issue since we always keep snapshots around
+    #   and don't run older migrations
```

Comment:
Would you mind adding a link to the issue in question where we discovered this and found the workaround? I can see this being a pain to lookup again, as took a while for us to understand before.

https://github.com/django-guardian/django-guardian/issues/281